code/model/modelread.cpp

```diff
@@ -4768,6 +4768,51 @@ void model_instance_global_to_local_dir(vec3d* out_dir, const vec3d* in_dir, con
 		delete[] submodelStack;
 }
 
+void model_instance_global_to_local_point_orient(vec3d* outpnt, matrix* outorient, const vec3d* submodel_pnt, const matrix* submodel_orient, const polymodel* pm, const polymodel_instance* pmi, int submodel_num, const matrix* objorient, const vec3d* objpos) {
+	Assert(pm->id == pmi->model_num);
+
+	constexpr int preallocatedStackDepth = 5;
+	std::tuple<const matrix*, const vec3d*, const vec3d*> preallocatedStack[preallocatedStackDepth];
+
+	auto submodelStack = pm->submodel[submodel_num].depth <= preallocatedStackDepth ? preallocatedStack : new std::tuple<const matrix*, const vec3d*, const vec3d*>[pm->submodel[submodel_num].depth];
+	int stackCounter = 0;
+
+	int mn = submodel_num;
+
+	//Go up the chain of parents to build a stack of transformations from parent -> child
+	while ((mn >= 0) && (pm->submodel[mn].parent >= 0)) {
```

Comment:
Should this chain have the `use_last_frame` check to select orient vs. prev_orient?

Reply:
Probably doesn't need it.
It's only included in the functions that explicitly need it for collision. All others don't, including the ones for the other direction.